### PR TITLE
[zh-cn] update 'Safely inserting external content into a page'

### PR DESCRIPTION
**Pull request by jasonren0403**

### CODE REVIEW

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 1]
```diff
@@ -1,70 +1,68 @@
...
 
-你还需要记住扩展有授权的上下文，例如后台的脚本 backgroup scrips 以及内容脚本 content scripts。最坏的情况里，一个内嵌的洁白你可能运行在一个或多种上下文中，导致一些提取操作。这种情况会导致用户的浏览器暴露在远程攻击下，被注入代码访问用户的重要数据，例如密码、浏览器历史或者浏览器行为等。
+你还需要记住，扩展具有特权上下文，例如后台脚本和内容脚本。最坏的情况里，一个内嵌的脚本可能运行在其中一个上下文中，这种情况被称为特权提升。在这种情况下，注入代码的网站可以访问关键的用户数据，如密码、浏览器历史记录或浏览行为，从而使用户的浏览器暴露在远程攻击的风险之下。
 
-这篇文章说明如何安全的对待远端数据并添加到 DOM。
+本文将探讨如何安全地处理远程数据并将其添加到 DOM 中。
 
 ## 处理任意字符串
 
 当处理字符串时，有很多推荐的策略来安全添加他们到页面：标准的 DOM 节点创建方法或 jQuery。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:21]:
```suggestion
当处理字符串时，有很多推荐的方法可以安全地将它们添加到页面：标准的 DOM 节点创建方法或 jQuery。
```

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 2]
```diff
@@ -78,31 +76,29 @@ An HTML sanitization library strips anything that could lead to script execution
...
+let cleanHTML = DOMPurify.sanitize(externalHTML, { SAFE_FOR_JQUERY: true });
 elem.html(cleanHTML);
 ```
 
 ### 模板引擎
 
-另一个常见的模式是对一个页面创建本地 HTML 模板并通过远端的值来填空。这种方法被广泛应用，应该注意去避免构造函数的使用，可能会导致执行代码的注入。当模板引擎使用构造函数插入 HTML 到文档的时候会发生这种情况，如果用来注入原始 HTML 的变量是远端代码，这就属于上面介绍中的安全风险。
-
-例如，当使用[mustache templates](https://mustache.github.io/) 你必须使用两个花括号`\{{variable}}`,来去掉任何 HTML。使用三个花括号`\{\{{variable}}}` 必须避免注入原始 HTML 字符串到模板。
+另一个常见的模式是对一个页面创建本地 HTML 模板并通过远端的值来填空。这种方法被广泛应用，应该注意去避免构建函数的使用，可能会导致执行代码的注入。当模板引擎使用构建函数插入未处理的 HTML 到文档的时候会发生这种情况。如果用来注入原始 HTML 的变量是远端代码，则会面临导言中提到的同样的安全风险。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:94]:
```suggestion
另一个常见的模式是对一个页面创建本地 HTML 模板并通过远端的值来填空。这种方法被广泛应用，应该注意去避免构建函数的使用，可能会导致执行代码的注入。当模板引擎使用构建函数插入未处理的 HTML 到文档的时候会发生这种情况。如果用来注入原始 HTML 的变量是远端代码，则会面临引言中提到的同样的安全风险。
```

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 3]
```diff
@@ -78,31 +76,29 @@ An HTML sanitization library strips anything that could lead to script execution
...
 
-[Handlebars](http://handlebarsjs.com/) 工作原理也是类似，`\{{variable}}` 。同时，如果创造了 Handlebars helper 使用 `Handlebars.SafeString` 使用`Handlebars.escapeExpression()` 来消除任何传给 helper 的动态参数。`Handlebars.SafeString` 的结果被认为是安全的，当插入到双 handlebars 的时候就不会做消除动作。
+例如，在使用 [mustache 模板](https://mustache.github.io/)时，必须使用双大括号，即 `\{{variable}}`，它可以转义任何 HTML。必须避免使用三重大括号（`\{\{{variable}}}`），因为这会注入原始 HTML 字符串，并可能在模板中添加可执行代码。[Handlebars](https://handlebarsjs.com/) 的工作方式与此类似，双大括号中的变量（`\{{variable}}`）会被转义。而三重大括号中的变量则是原始变量，必须避免使用。此外，如果你使用 `Handlebars.SafeString` 创建了一个 Handlebars 辅助函数，请使用 `Handlebars.escapeExpression()` 来转义传递给辅助函数的任何动态参数。之所以需要这样做，是因为 `Handlebars.SafeString` 产生的变量被认为是安全的，而且在插入双大括号中时不会转义。
 
-在其他模板系统中也有类似的构造函数，需要有同样的考虑。
+在其他模板系统中也有类似的构建函数，需要有同样的考虑。
 
-## 其他参考
+## 补充阅读
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:100]: 扩展阅读？

**Review by yin1999 — APPROVED** [review]
一些小建议

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 4]
```diff
@@ -78,31 +76,29 @@ An HTML sanitization library strips anything that could lead to script execution
...
 
 关于这个主题更多的信息，可以查看下面的文章
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:102]:
```suggestion
关于这个主题更多的信息，可以查看以下文章：
```

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 5]
```diff
@@ -1,70 +1,68 @@
...
+let data = JSON.parse(responseText);
+let div = document.createElement("div");
 div.className = data.className;
-div.textContent = "Your favorite color is now " + data.color;
+div.textContent = `你最喜欢的颜色现在是${data.color}`;
 addonElement.appendChild(div);
 ```
 
-这种方法安全原因是使用`.textContent` 时会自动消除 `data.color`.中的任何远程 HTML 代码。
+这种方法安全原因是使用 `.textContent` 时会自动转义 `data.color` 中的任何远程 HTML 代码。
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:36]:
```suggestion
这种方法安全的原因是使用 `.textContent` 时会自动转义 `data.color` 中的任何远程 HTML 代码。
```

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 6]
```diff
@@ -1,70 +1,68 @@
...
 当处理字符串时，有很多推荐的策略来安全添加他们到页面：标准的 DOM 节点创建方法或 jQuery。
 
-### DOM 节点创建方法/函数
+### DOM 节点创建方法
 
-一个轻量级的方法来插入字符串到页面是使用原生的 DOM 操纵方法/函数：[`document.createElement`](/zh-CN/docs/Web/API/Document/createElement)、[`Element.setAttribute`](/zh-CN/docs/Web/API/Element/setAttribute)，以及[`Node.textContent`](/zh-CN/docs/Web/API/Node/textContent)。安全的方法是分别创建节点并使用 textContent 属性赋值：
+一个轻量级的方法来插入字符串到页面是使用原生的 DOM 操纵方法：[`document.createElement`](/zh-CN/docs/Web/API/Document/createElement)、[`Element.setAttribute`](/zh-CN/docs/Web/API/Element/setAttribute) 以及[`Node.textContent`](/zh-CN/docs/Web/API/Node/textContent)。安全的方法是分别创建节点并使用 textContent 属性赋值：
 
 ```js example-good
 var data = JSON.parse(responseText);
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:28]:
```suggestion
```

File: files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md [hunk 7]
```diff
@@ -1,70 +1,68 @@
...
 
 ## 处理任意字符串
 
 当处理字符串时，有很多推荐的策略来安全添加他们到页面：标准的 DOM 节点创建方法或 jQuery。
 
-### DOM 节点创建方法/函数
+### DOM 节点创建方法
 
-一个轻量级的方法来插入字符串到页面是使用原生的 DOM 操纵方法/函数：[`document.createElement`](/zh-CN/docs/Web/API/Document/createElement)、[`Element.setAttribute`](/zh-CN/docs/Web/API/Element/setAttribute)，以及[`Node.textContent`](/zh-CN/docs/Web/API/Node/textContent)。安全的方法是分别创建节点并使用 textContent 属性赋值：
+一个轻量级的方法来插入字符串到页面是使用原生的 DOM 操纵方法：[`document.createElement`](/zh-CN/docs/Web/API/Document/createElement)、[`Element.setAttribute`](/zh-CN/docs/Web/API/Element/setAttribute) 以及[`Node.textContent`](/zh-CN/docs/Web/API/Node/textContent)。安全的方法是分别创建节点并使用 textContent 属性赋值：
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/safely_inserting_external_content_into_a_page/index.md:25]:
```suggestion
一个轻量级的方法来插入字符串到页面是使用原生的 DOM 操纵方法：[`document.createElement`](/zh-CN/docs/Web/API/Document/createElement)、[`Element.setAttribute`](/zh-CN/docs/Web/API/Element/setAttribute) 以及 [`Node.textContent`](/zh-CN/docs/Web/API/Node/textContent)。安全的方法是分别创建节点并使用 textContent 属性赋值：
```

**Pull request merged**